### ISSUE TITLE
Table is passed as userdata

### ISSUE THREAD

**Issue by nrother** [nrother/dynamiclua#5]
This code will fail:

``` csharp
dynamic tab = lua("return {a = 1, b = 2, c = 3}");
lua.tab = tab;
lua("print(type(tab))");
lua("for k,v in pairs(tab) do print(k,v) end");
```

Error message:

```
[string "chunk"]:1: bad argument #1 to 'pairs' (table expected, got userdata)
```

Thanks for Kees van Marle for reporting this.

**Comment by nrother** [comment]
This is caused by tab being a DynamicArray instead of the expected DynamicLuaTable.

**Comment by nrother** [comment]
Fixed in 9450545a4bd064e021f2eba1fc69a43d6f60f770
